### DISCUSSION TITLE
How to check if a point is within a object? The "containsPoint" method appears to have some issues.

### DISCUSSION THREAD

**Discussion by olivia1520** [fabricjs/fabric.js#10000] [Q&A]
I noticed that 'containsPoint' returns true as long as the mouse cursor is within the bounding box of an object. Is there a method to determine if a point truly lies inside a polygon, rather than just within its bounding box?

`<script src="https://unpkg.com/fabric@5.3.0/dist/fabric.min.js"></script>
</head>
<body>
    <canvas id="canvas" width="550" height="580" style="border: 1px solid rgb(224, 223, 223)"></canvas>
</body>
<script>
    var canvas = new fabric.Canvas('canvas');
    fabric.Object.prototype.originX = "center"; 
    fabric.Object.prototype.originY = "center";

    var points= [
            {
                "x": 44.844926350523174,
                "y": 180.83672060444002
            },
            {
                "x": 158.07899045597657,
                "y": 149.93274821438393
            },
            {
                "x": 188.61401897879543,
                "y": 268.46853272418804
            },
            {
                "x": 119.06200956570797,
                "y": 271.43192733693314
            }
        ]

    var polygon = new fabric.Polygon(points,{
        fill: "transparent", 
        stroke: "red", 
        strokeWidth: 1,
        //hasBorders: false,
        hasControls:false,
        //selectable:false,
        hoverCursor: 'default',
        originX:"center",
        originY:"center",
        lockMovementX:true,
        lockMovementY:true
    })

    canvas.add(polygon);
    canvas.on("mouse:down",function(options){
        let pointer = canvas.getPointer(options.e);
        console.log("pointer:", pointer);
        console.log("contains ",polygon.containsPoint(pointer));
    })
</script>`
![image](https://github.com/user-attachments/assets/2d47a641-818c-412c-ae42-1ee144dd9138)

**Comment by asturur** [comment]
first of all it depends if you are working with scene coordinates or canvas coordinates.
Fabric 6 has new utility called 'isPointInPolygon' that will let you do that after you transformed the points accordingly.
For 5.3  you will have to look at findCrossPoint and write a custom method for the polygon class.
